File: server/internal/models/settlement_price.go
package models

import (
	"database/sql/driver"
	"encoding/json"
	"fmt"
	"time"
)

// ============================================================
// 结算价相关常量
// ============================================================

// ChangeType 变更类型
type ChangeType int16

const (
	ChangeTypeInit       ChangeType = 1 // 初始化（从模板创建）
	ChangeTypeRate       ChangeType = 2 // 费率调整
	ChangeTypeDeposit    ChangeType = 3 // 押金返现调整
	ChangeTypeSim        ChangeType = 4 // 流量费返现调整
	ChangeTypeActivation ChangeType = 5 // 激活奖励调整
	ChangeTypeBatch      ChangeType = 6 // 批量调整
	ChangeTypeSync       ChangeType = 7 // 模板同步
)

// ChangeTypeName 获取变更类型名称
func ChangeTypeName(ct ChangeType) string {
	switch ct {
	case ChangeTypeInit:
		return "初始化"
	case ChangeTypeRate:
		return "费率调整"
	case ChangeTypeDeposit:
		return "押金返现调整"
	case ChangeTypeSim:
		return "流量费返现调整"
	case ChangeTypeActivation:
		return "激活奖励调整"
	case ChangeTypeBatch:
		return "批量调整"
	case ChangeTypeSync:
		return "模板同步"
	default:
		return "未知类型"
	}
}

// ConfigType 配置类型
type ConfigType int16

const (
	ConfigTypeSettlement ConfigType = 1 // 结算价
	ConfigTypeReward     ConfigType = 2 // 奖励配置
)

// ConfigTypeName 获取配置类型名称
func ConfigTypeName(ct ConfigType) string {
	switch ct {
	case ConfigTypeSettlement:
		return "结算价"
	case ConfigTypeReward:
		return "奖励配置"
	default:
		return "未知类型"
	}
}

// OperatorType 操作者类型
type OperatorType int16

const (
	OperatorTypeAdmin OperatorType = 1 // 管理员
	OperatorTypeAgent OperatorType = 2 // 代理商
)

// ============================================================
// 押金返现配置JSONB类型
// ============================================================

// DepositCashbackItem 押金返现配置项
type DepositCashbackItem struct {
	DepositAmount  int64 `json:"deposit_amount"`  // 押金金额（分）
	CashbackAmount int64 `json:"cashback_amount"` // 返现金额（分）
}

// DepositCashbacks 押金返现配置列表
type DepositCashbacks []DepositCashbackItem

// Scan 实现sql.Scanner接口
func (d *DepositCashbacks) Scan(value interface{}) error {
	if value == nil {
		*d = make(DepositCashbacks, 0)
		return nil
	}

	var bytes []byte
	switch v := value.(type) {
	case []byte:
		bytes = v
	case string:
		bytes = []byte(v)
	default:
		return fmt.Errorf("cannot scan type %T into DepositCashbacks", value)
	}

	if len(bytes) == 0 || string(bytes) == "[]" {
		*d = make(DepositCashbacks, 0)
		return nil
	}

	return json.Unmarshal(bytes, d)
}

// Value 实现driver.Valuer接口
func (d DepositCashbacks) Value() (driver.Value, error) {
	if d == nil {
		return "[]", nil
	}
	return json.Marshal(d)
}

// ============================================================
// 激活奖励配置JSONB类型
// ============================================================

// ActivationRewardItem 激活奖励配置项
type ActivationRewardItem struct {
	RewardName      string `json:"reward_name"`       // 奖励名称
	MinRegisterDays int    `json:"min_register_days"` // 最小入网天数
	MaxRegisterDays int    `json:"max_register_days"` // 最大入网天数
	TargetAmount    int64  `json:"target_amount"`     // 目标交易量（分）
	RewardAmount    int64  `json:"reward_amount"`     // 奖励金额（分）
	Priority        int    `json:"priority"`          // 优先级
}

// ActivationRewards 激活奖励配置列表
type ActivationRewards []ActivationRewardItem

// Scan 实现sql.Scanner接口
func (a *ActivationRewards) Scan(value interface{}) error {
	if value == nil {
		*a = make(ActivationRewards, 0)
		return nil
	}

	var bytes []byte
	switch v := value.(type) {
	case []byte:
		bytes = v
	case string:
		bytes = []byte(v)
	default:
		return fmt.Errorf("cannot scan type %T into ActivationRewards", value)
	}

	if len(bytes) == 0 || string(bytes) == "[]" {
		*a = make(ActivationRewards, 0)
		return nil
	}

	return json.Unmarshal(bytes, a)
}

// Value 实现driver.Valuer接口
func (a ActivationRewards) Value() (driver.Value, error) {
	if a == nil {
		return "[]", nil
	}
	return json.Marshal(a)
}

// ============================================================
// 通道结算价表
// ============================================================

// SettlementPrice 通道结算价
type SettlementPrice struct {
	ID        int64  `json:"id" gorm:"primaryKey"`
	AgentID   int64  `json:"agent_id" gorm:"not null;index"`
	ChannelID int64  `json:"channel_id" gorm:"not null;index"`
	TemplateID *int64 `json:"template_id" gorm:"index"`
	BrandCode string `json:"brand_code" gorm:"size:32;default:''"`

	// 费率配置（动态JSONB）
	RateConfigs RateConfigs `json:"rate_configs" gorm:"type:jsonb;default:'{}'"`

	// 旧字段保留兼容
	CreditRate   *string `json:"credit_rate" gorm:"type:decimal(10,4)"`
	DebitRate    *string `json:"debit_rate" gorm:"type:decimal(10,4)"`
	DebitCap     *string `json:"debit_cap" gorm:"type:decimal(10,2)"`
	UnionpayRate *string `json:"unionpay_rate" gorm:"type:decimal(10,4)"`
	WechatRate   *string `json:"wechat_rate" gorm:"type:decimal(10,4)"`
	AlipayRate   *string `json:"alipay_rate" gorm:"type:decimal(10,4)"`

	// 押金返现配置（JSONB数组）
	DepositCashbacks DepositCashbacks `json:"deposit_cashbacks" gorm:"type:jsonb;default:'[]'"`

	// 流量费返现配置
	SimFirstCashback     int64 `json:"sim_first_cashback" gorm:"default:0"`
	SimSecondCashback    int64 `json:"sim_second_cashback" gorm:"default:0"`
	SimThirdPlusCashback int64 `json:"sim_third_plus_cashback" gorm:"default:0"`

	// 元数据
	Version     int        `json:"version" gorm:"default:1"`
	Status      int16      `json:"status" gorm:"default:1"`
	EffectiveAt time.Time  `json:"effective_at" gorm:"default:now()"`
	CreatedAt   time.Time  `json:"created_at" gorm:"default:now()"`
	UpdatedAt   time.Time  `json:"updated_at" gorm:"default:now()"`
	CreatedBy   *int64     `json:"created_by"`
	UpdatedBy   *int64     `json:"updated_by"`
}

// TableName 表名
func (SettlementPrice) TableName() string {
	return "settlement_prices"
}

// ============================================================
// 代理商奖励配置表
// ============================================================

// AgentRewardSetting 代理商奖励配置
type AgentRewardSetting struct {
	ID         int64  `json:"id" gorm:"primaryKey"`
	AgentID    int64  `json:"agent_id" gorm:"not null;uniqueIndex"`
	TemplateID *int64 `json:"template_id" gorm:"index"`

	// 奖励金额（差额分配模式）
	RewardAmount int64 `json:"reward_amount" gorm:"default:0"`

	// 激活奖励配置（JSONB数组）
	ActivationRewards ActivationRewards `json:"activation_rewards" gorm:"type:jsonb;default:'[]'"`

	// 元数据
	Version     int        `json:"version" gorm:"default:1"`
	Status      int16      `json:"status" gorm:"default:1"`
	EffectiveAt time.Time  `json:"effective_at" gorm:"default:now()"`
	CreatedAt   time.Time  `json:"created_at" gorm:"default:now()"`
	UpdatedAt   time.Time  `json:"updated_at" gorm:"default:now()"`
	CreatedBy   *int64     `json:"created_by"`
	UpdatedBy   *int64     `json:"updated_by"`
}

// TableName 表名
func (AgentRewardSetting) TableName() string {
	return "agent_reward_settings"
}

// ============================================================
// 调价记录表
// ============================================================

// PriceChangeLog 调价记录
type PriceChangeLog struct {
	ID                int64  `json:"id" gorm:"primaryKey"`
	AgentID           int64  `json:"agent_id" gorm:"not null;index"`
	ChannelID         *int64 `json:"channel_id" gorm:"index"`
	SettlementPriceID *int64 `json:"settlement_price_id" gorm:"index"`
	RewardSettingID   *int64 `json:"reward_setting_id" gorm:"index"`

	// 变更类型
	ChangeType ChangeType `json:"change_type" gorm:"not null"`
	ConfigType ConfigType `json:"config_type" gorm:"not null"`

	// 变更内容
	FieldName     string `json:"field_name" gorm:"size:100"`
	OldValue      string `json:"old_value" gorm:"type:text"`
	NewValue      string `json:"new_value" gorm:"type:text"`
	ChangeSummary string `json:"change_summary" gorm:"size:500"`

	// 完整快照
	SnapshotBefore JSONMap `json:"snapshot_before" gorm:"type:jsonb"`
	SnapshotAfter  JSONMap `json:"snapshot_after" gorm:"type:jsonb"`

	// 操作信息
	OperatorType OperatorType `json:"operator_type" gorm:"default:1"`
	OperatorID   int64        `json:"operator_id" gorm:"not null;index"`
	OperatorName string       `json:"operator_name" gorm:"size:100"`
	Source       string       `json:"source" gorm:"size:20;default:'PC'"`
	IPAddress    string       `json:"ip_address" gorm:"size:50"`
	UserAgent    string       `json:"user_agent" gorm:"size:500"`

	CreatedAt time.Time `json:"created_at" gorm:"default:now();index"`
}

// TableName 表名
func (PriceChangeLog) TableName() string {
	return "price_change_logs"
}

// JSONMap 通用JSON Map类型
type JSONMap map[string]interface{}

// Scan 实现sql.Scanner接口
func (j *JSONMap) Scan(value interface{}) error {
	if value == nil {
		*j = make(JSONMap)
		return nil
	}

	var bytes []byte
	switch v := value.(type) {
	case []byte:
		bytes = v
	case string:
		bytes = []byte(v)
	default:
		return fmt.Errorf("cannot scan type %T into JSONMap", value)
	}

	if len(bytes) == 0 || string(bytes) == "{}" {
		*j = make(JSONMap)
		return nil
	}

	return json.Unmarshal(bytes, j)
}

// Value 实现driver.Valuer接口
func (j JSONMap) Value() (driver.Value, error) {
	if j == nil {
		return "{}", nil
	}
	return json.Marshal(j)
}

// ============================================================
// 请求/响应DTO
// ============================================================

// CreateSettlementPriceRequest 创建结算价请求
type CreateSettlementPriceRequest struct {
	AgentID    int64  `json:"agent_id" binding:"required"`
	ChannelID  int64  `json:"channel_id" binding:"required"`
	TemplateID *int64 `json:"template_id"`
	BrandCode  string `json:"brand_code"`
}

// UpdateRateRequest 更新费率请求
type UpdateRateRequest struct {
	RateConfigs  RateConfigs `json:"rate_configs"`
	CreditRate   *string     `json:"credit_rate"`
	DebitRate    *string     `json:"debit_rate"`
	DebitCap     *string     `json:"debit_cap"`
	UnionpayRate *string     `json:"unionpay_rate"`
	WechatRate   *string     `json:"wechat_rate"`
	AlipayRate   *string     `json:"alipay_rate"`
}

// UpdateDepositCashbackRequest 更新押金返现请求
type UpdateDepositCashbackRequest struct {
	DepositCashbacks DepositCashbacks `json:"deposit_cashbacks" binding:"required"`
}

// UpdateSimCashbackRequest 更新流量费返现请求
type UpdateSimCashbackRequest struct {
	SimFirstCashback     int64 `json:"sim_first_cashback"`
	SimSecondCashback    int64 `json:"sim_second_cashback"`
	SimThirdPlusCashback int64 `json:"sim_third_plus_cashback"`
}

// UpdateActivationRewardRequest 更新激活奖励请求
type UpdateActivationRewardRequest struct {
	ActivationRewards ActivationRewards `json:"activation_rewards" binding:"required"`
}

// SettlementPriceListRequest 结算价列表请求
type SettlementPriceListRequest struct {
	AgentID   *int64 `form:"agent_id"`
	ChannelID *int64 `form:"channel_id"`
	Status    *int16 `form:"status"`
	Page      int    `form:"page,default=1"`
	PageSize  int    `form:"page_size,default=20"`
}

// SettlementPriceListResponse 结算价列表响应
type SettlementPriceListResponse struct {
	List  []SettlementPriceItem `json:"list"`
	Total int64                 `json:"total"`
	Page  int                   `json:"page"`
	Size  int                   `json:"size"`
}

// SettlementPriceItem 结算价列表项
type SettlementPriceItem struct {
	ID          int64            `json:"id"`
	AgentID     int64            `json:"agent_id"`
	AgentName   string           `json:"agent_name"`
	ChannelID   int64            `json:"channel_id"`
	ChannelName string           `json:"channel_name"`
	BrandCode   string           `json:"brand_code"`
	RateConfigs RateConfigs      `json:"rate_configs"`
	DepositCashbacks DepositCashbacks `json:"deposit_cashbacks"`
	SimFirstCashback     int64  `json:"sim_first_cashback"`
	SimSecondCashback    int64  `json:"sim_second_cashback"`
	SimThirdPlusCashback int64  `json:"sim_third_plus_cashback"`
	Version     int              `json:"version"`
	Status      int16            `json:"status"`
	EffectiveAt time.Time        `json:"effective_at"`
	UpdatedAt   time.Time        `json:"updated_at"`
}

// PriceChangeLogListRequest 调价记录列表请求
type PriceChangeLogListRequest struct {
	AgentID    *int64      `form:"agent_id"`
	ChannelID  *int64      `form:"channel_id"`
	ChangeType *ChangeType `form:"change_type"`
	ConfigType *ConfigType `form:"config_type"`
	StartDate  string      `form:"start_date"`
	EndDate    string      `form:"end_date"`
	Page       int         `form:"page,default=1"`
	PageSize   int         `form:"page_size,default=20"`
}

// PriceChangeLogListResponse 调价记录列表响应
type PriceChangeLogListResponse struct {
	List  []PriceChangeLogItem `json:"list"`
	Total int64                `json:"total"`
	Page  int                  `json:"page"`
	Size  int                  `json:"size"`
}

// PriceChangeLogItem 调价记录列表项
type PriceChangeLogItem struct {
	ID             int64      `json:"id"`
	AgentID        int64      `json:"agent_id"`
	AgentName      string     `json:"agent_name"`
	ChannelID      *int64     `json:"channel_id"`
	ChannelName    string     `json:"channel_name"`
	ChangeType     ChangeType `json:"change_type"`
	ChangeTypeName string     `json:"change_type_name"`
	ConfigType     ConfigType `json:"config_type"`
	ConfigTypeName string     `json:"config_type_name"`
	FieldName      string     `json:"field_name"`
	OldValue       string     `json:"old_value"`
	NewValue       string     `json:"new_value"`
	ChangeSummary  string     `json:"change_summary"`
	OperatorName   string     `json:"operator_name"`
	Source         string     `json:"source"`
	CreatedAt      time.Time  `json:"created_at"`
}

// AgentRewardSettingRequest 代理商奖励配置请求
type AgentRewardSettingRequest struct {
	AgentID      int64  `json:"agent_id" binding:"required"`
	TemplateID   *int64 `json:"template_id"`
	RewardAmount int64  `json:"reward_amount"`
}

// AgentRewardSettingResponse 代理商奖励配置响应
type AgentRewardSettingResponse struct {
	ID                int64             `json:"id"`
	AgentID           int64             `json:"agent_id"`
	AgentName         string            `json:"agent_name"`
	TemplateID        *int64            `json:"template_id"`
	TemplateName      string            `json:"template_name"`
	RewardAmount      int64             `json:"reward_amount"`
	ActivationRewards ActivationRewards `json:"activation_rewards"`
	Version           int               `json:"version"`
	Status            int16             `json:"status"`
	EffectiveAt       time.Time         `json:"effective_at"`
	UpdatedAt         time.Time         `json:"updated_at"`
}

File: web/src/api/settlementPrice.ts
import { get, post, put } from './request'

// ============================================================
// 结算价类型定义
// ============================================================

/** 费率配置 */
export interface RateConfig {
  rate: string
}

/** 费率配置集合 */
export interface RateConfigs {
  [key: string]: RateConfig
}

/** 押金返现配置项 */
export interface DepositCashbackItem {
  deposit_amount: number
  cashback_amount: number
}

/** 激活奖励配置项 */
export interface ActivationRewardItem {
  reward_name: string
  min_register_days: number
  max_register_days: number
  target_amount: number
  reward_amount: number
  priority: number
}

/** 结算价 */
export interface SettlementPrice {
  id: number
  agent_id: number
  agent_name?: string
  channel_id: number
  channel_name?: string
  template_id?: number
  brand_code: string
  rate_configs: RateConfigs
  credit_rate?: string
  debit_rate?: string
  debit_cap?: string
  unionpay_rate?: string
  wechat_rate?: string
  alipay_rate?: string
  deposit_cashbacks: DepositCashbackItem[]
  sim_first_cashback: number
  sim_second_cashback: number
  sim_third_plus_cashback: number
  high_rate_configs: HighRateConfigs
  d0_extra_configs: D0ExtraConfigs
  version: number
  status: number
  effective_at: string
  created_at: string
  updated_at: string
}

/** 结算价列表项 */
export interface SettlementPriceItem {
  id: number
  agent_id: number
  agent_name: string
  channel_id: number
  channel_name: string
  brand_code: string
  rate_configs: RateConfigs
  deposit_cashbacks: DepositCashbackItem[]
  sim_first_cashback: number
  sim_second_cashback: number
  sim_third_plus_cashback: number
  high_rate_configs: HighRateConfigs
  d0_extra_configs: D0ExtraConfigs
  version: number
  status: number
  effective_at: string
  updated_at: string
}

/** 结算价列表响应 */
export interface SettlementPriceListResponse {
  list: SettlementPriceItem[]
  total: number
  page: number
  size: number
}

/** 结算价列表请求参数 */
export interface SettlementPriceListParams {
  agent_id?: number
  channel_id?: number
  status?: number
  page?: number
  page_size?: number
}

/** 创建结算价请求 */
export interface CreateSettlementPriceRequest {
  agent_id: number
  channel_id: number
  template_id?: number
  brand_code?: string
}

/** 更新费率请求 */
export interface UpdateRateRequest {
  rate_configs?: RateConfigs
  credit_rate?: string
  debit_rate?: string
  debit_cap?: string
  unionpay_rate?: string
  wechat_rate?: string
  alipay_rate?: string
}

/** 更新押金返现请求 */
export interface UpdateDepositCashbackRequest {
  deposit_cashbacks: DepositCashbackItem[]
}

/** 更新流量费返现请求 */
export interface UpdateSimCashbackRequest {
  sim_first_cashback: number
  sim_second_cashback: number
  sim_third_plus_cashback: number
}

/** 高调费率配置项 */
export interface HighRateConfig {
  rate: string  // 高调费率（%）
}

/** 高调费率配置集合 */
export interface HighRateConfigs {
  [rateType: string]: HighRateConfig
}

/** P+0加价配置项 */
export interface D0ExtraConfig {
  extra_fee: number  // 加价金额（分）
}

/** P+0加价配置集合 */
export interface D0ExtraConfigs {
  [rateType: string]: D0ExtraConfig
}

/** 更新高调费率请求 */
export interface UpdateHighRateRequest {
  high_rate_configs: HighRateConfigs
}

/** 更新P+0加价请求 */
export interface UpdateD0ExtraRequest {
  d0_extra_configs: D0ExtraConfigs
}

// ============================================================
// 代理商奖励配置类型定义
// ============================================================

/** 代理商奖励配置 */
export interface AgentRewardSetting {
  id: number
  agent_id: number
  agent_name?: string
  template_id?: number
  template_name?: string
  reward_amount: number
  activation_rewards: ActivationRewardItem[]
  version: number
  status: number
  effective_at: string
  created_at: string
  updated_at: string
}

/** 创建代理商奖励配置请求 */
export interface CreateAgentRewardSettingRequest {
  agent_id: number
  template_id?: number
  reward_amount?: number
}

/** 更新激活奖励请求 */
export interface UpdateActivationRewardRequest {
  activation_rewards: ActivationRewardItem[]
}

// ============================================================
// 调价记录类型定义
// ============================================================

/** 变更类型 */
export enum ChangeType {
  Init = 1,       // 初始化
  Rate = 2,       // 费率调整
  Deposit = 3,    // 押金返现调整
  Sim = 4,        // 流量费返现调整
  Activation = 5, // 激活奖励调整
  Batch = 6,      // 批量调整
  Sync = 7,       // 模板同步
}

/** 配置类型 */
export enum ConfigType {
  Settlement = 1, // 结算价
  Reward = 2,     // 奖励配置
}

/** 调价记录 */
export interface PriceChangeLog {
  id: number
  agent_id: number
  agent_name: string
  channel_id?: number
  channel_name: string
  change_type: ChangeType
  change_type_name: string
  config_type: ConfigType
  config_type_name: string
  field_name: string
  old_value: string
  new_value: string
  change_summary: string
  operator_name: string
  source: string
  created_at: string
}

/** 调价记录列表响应 */
export interface PriceChangeLogListResponse {
  list: PriceChangeLog[]
  total: number
  page: number
  size: number
}

/** 调价记录列表请求参数 */
export interface PriceChangeLogListParams {
  agent_id?: number
  channel_id?: number
  change_type?: ChangeType
  config_type?: ConfigType
  start_date?: string
  end_date?: string
  page?: number
  page_size?: number
}

// ============================================================
// 结算价API
// ============================================================

/**
 * 获取结算价列表
 */
export function getSettlementPrices(
  params: SettlementPriceListParams
): Promise<SettlementPriceListResponse> {
  return get<SettlementPriceListResponse>('/v1/settlement-prices', params)
}

/**
 * 获取结算价详情
 */
export function getSettlementPrice(id: number): Promise<SettlementPrice> {
  return get<SettlementPrice>(`/v1/settlement-prices/${id}`)
}

/**
 * 创建结算价
 */
export function createSettlementPrice(
  data: CreateSettlementPriceRequest
): Promise<SettlementPrice> {
  return post<SettlementPrice>('/v1/settlement-prices', data)
}

/**
 * 更新费率
 */
export function updateSettlementPriceRate(
  id: number,
  data: UpdateRateRequest
): Promise<SettlementPrice> {
  return put<SettlementPrice>(`/v1/settlement-prices/${id}/rate`, data)
}

/**
 * 更新押金返现
 */
export function updateSettlementPriceDeposit(
  id: number,
  data: UpdateDepositCashbackRequest
): Promise<SettlementPrice> {
  return put<SettlementPrice>(`/v1/settlement-prices/${id}/deposit`, data)
}

/**
 * 更新流量费返现
 */
export function updateSettlementPriceSim(
  id: number,
  data: UpdateSimCashbackRequest
): Promise<SettlementPrice> {
  return put<SettlementPrice>(`/v1/settlement-prices/${id}/sim`, data)
}

/**
 * 获取结算价调价记录
 */
export function getSettlementPriceChangeLogs(
  id: number,
  params?: { page?: number; page_size?: number }
): Promise<PriceChangeLogListResponse> {
  return get<PriceChangeLogListResponse>(`/v1/settlement-prices/${id}/change-logs`, params)
}

/**
 * 更新高调费率配置
 */
export function updateSettlementPriceHighRate(
  id: number,
  data: UpdateHighRateRequest
): Promise<SettlementPrice> {
  return put<SettlementPrice>(`/v1/settlement-prices/${id}/high-rate`, data)
}

/**
 * 更新P+0加价配置
 */
export function updateSettlementPriceD0Extra(
  id: number,
  data: UpdateD0ExtraRequest
): Promise<SettlementPrice> {
  return put<SettlementPrice>(`/v1/settlement-prices/${id}/d0-extra`, data)
}

// ============================================================
// 代理商奖励配置API
// ============================================================

/**
 * 获取奖励配置列表
 */
export function getRewardSettings(params?: {
  page?: number
  page_size?: number
}): Promise<{ list: AgentRewardSetting[]; total: number; page: number; size: number }> {
  return get('/v1/reward-settings', params)
}

/**
 * 获取奖励配置详情
 */
export function getRewardSetting(id: number): Promise<AgentRewardSetting> {
  return get<AgentRewardSetting>(`/v1/reward-settings/${id}`)
}

/**
 * 创建奖励配置
 */
export function createRewardSetting(
  data: CreateAgentRewardSettingRequest
): Promise<AgentRewardSetting> {
  return post<AgentRewardSetting>('/v1/reward-settings', data)
}

/**
 * 更新激活奖励
 */
export function updateRewardSettingActivation(
  id: number,
  data: UpdateActivationRewardRequest
): Promise<AgentRewardSetting> {
  return put<AgentRewardSetting>(`/v1/reward-settings/${id}/activation`, data)
}

/**
 * 获取奖励配置调价记录
 */
export function getRewardSettingChangeLogs(
  id: number,
  params?: { page?: number; page_size?: number }
): Promise<PriceChangeLogListResponse> {
  return get<PriceChangeLogListResponse>(`/v1/reward-settings/${id}/change-logs`, params)
}

// ============================================================
// 调价记录API
// ============================================================

/**
 * 获取调价记录列表
 */
export function getPriceChangeLogs(
  params: PriceChangeLogListParams
): Promise<PriceChangeLogListResponse> {
  return get<PriceChangeLogListResponse>('/v1/price-change-logs', params)
}

/**
 * 获取调价记录详情
 */
export function getPriceChangeLog(id: number): Promise<PriceChangeLog> {
  return get<PriceChangeLog>(`/v1/price-change-logs/${id}`)
}

/**
 * 按代理商获取调价记录
 */
export function getAgentPriceChangeLogs(
  agentId: number,
  params?: { page?: number; page_size?: number }
): Promise<PriceChangeLogListResponse> {
  return get<PriceChangeLogListResponse>(`/v1/agents/${agentId}/price-change-logs`, params)
}
